labonneboite/web/static/js/recruiter-forms.js

```diff
@@ -0,0 +1,166 @@
+    if($('.identification-form').length > 0) {
+      var $form = $('.identification-form');
+
+      var rawValues = localStorage.getItem(FORM_STORAGE_KEY);
+      if(!rawValues) return;
+      try {
+        var values = JSON.parse(rawValues);
+
+        // Do not override value if the siret (coming from URL parameter) is already here
+        var $siret = $form.find("input[name='siret']");
+        if(!$siret.val()) $siret.val(values.siret);
+
+        $form.find("input[name='last_name']").val(values.lastName);
+        $form.find("input[name='first_name']").val(values.firstName);
+        $form.find("input[name='phone']").val(values.phone);
+        $form.find("input[name='email']").val(values.email);
+      } catch(error) {
+        console.log(error)
+        return;
+      }
+
+    }
+  }
+
+  // JOB AUTO-COMPLETE
+  var $newJobTable = $('.add-new-jobs table');
+  var $newJobTbody = $newJobTable.find('tbody');
+  var $newJobInput = $('.add-new-jobs input[name=new-job]');
+  var jobRowTemplate = '<tr id="{ROME}"><td><span><label for="{ROME}">{LABEL}</label></span></td>' +
+      '<td class="text-center"><span class="sr-only">Intéressé par les candidatures</span><input checked="checked" type="checkbox" name="{ROME}" value="lbb"></td>' +
+      '<td class="text-center"><span class="sr-only">Ouvert aux contrats d\'alternance</span><input checked="checked" type="checkbox" name="{ROME}" value="lba"></td>' +
+      '<td class="text-center"><button class="btn remove" data-rome="{ROME}">Supprimer</button></td></tr>';
+
+  $newJobInput.autocomplete({
+    delay: 200,
+    html: true,
+    minLength: 2,
+    position: {my: "left top", at: "left bottom"},
+    open: function (event, ui) {
+      $('.ui-autocomplete').off('menufocus hover mouseover mouseenter');
+    },
+    source: "/suggest_job_labels",
+    select: function (event, ui) {
+      addJob(ui.item.id, ui.item.label)
+    },
+    close: function(event, ui) {
+      $newJobInput.val('');
+    }
+  });
+
+  function addJob(rome, label) {
+    // Handle if job already exists
+    var alreadyExists = $('form tr#' + rome).length !== 0;
+    if(alreadyExists) {
+      alert('Le métier "' + label + '" est déjà présent.');
+      return;
+    }
+
+    var rowHTML = t(jobRowTemplate, { ROME: rome, LABEL: label })
+    $newJobTbody.append($(rowHTML));
+    $newJobTable.removeClass("hidden");
+  }
+
+
+  // Avoid submitting form when pressing 'enter'
+  $newJobInput.keydown(function(event) {
+    if (event.keyCode == 13) event.preventDefault();
+  });
+
+  // Remove added job
+  $(document).on('click', '.add-new-jobs table button.remove', function(event) {
+    event.preventDefault();
+    var rome = $(event.target).attr('data-rome');
+    $('form tr#' + rome).remove();
+
+    if($newJobTbody.find('tr').length === 0) $newJobTable.addClass("hidden");
+  });
+
+  // http://mir.aculo.us/2011/03/09/little-helpers-a-tweet-sized-javascript-templating-engine/
+  function t(s,d){
```

Comment:
Je veux bien avoir le code complet de cette fonction, du coup, avec des vrais noms. Du genre : 

function render_template(s, values) {
    for (var value in values) {
        ...
    }
}
```

c'est mon côté old school.

Reply:
:+1: